code/game/machinery/suit_modifier.dm

```diff
@@ -194,93 +244,129 @@
+		if(oldsuit)
+			guy.equip_to_slot(new chosen_suit, slot_wear_suit)
+			qdel(oldsuit)
+			guy.update_inv_wear_suit()
+		var/obj/item/clothing/head/helmet/space/oldhelmet = guy.get_item_by_slot(slot_head)
+		if(oldhelmet)
+			guy.equip_to_slot(new chosen_helmet, slot_head)
+			qdel(oldhelmet)
+			guy.update_inv_head()
 	unlock_atom(guy)
+	finished_animation()
 	use_power = MACHINE_POWER_USE_IDLE
 	activated = FALSE
 
 /obj/machinery/suit_modifier/proc/process_module_installation(var/mob/living/carbon/human/H)
 	if(activated)
 		return
-	lock_atom(H)
+	var/obj/item/clothing/suit/space/rig/R = H.is_wearing_item(/obj/item/clothing/suit/space/rig, slot_wear_suit)
+	if(!R)
+		return
 	activated = TRUE
 	use_power = MACHINE_POWER_USE_ACTIVE
-	activation_animation()
-	var/obj/item/clothing/suit/space/rig/R = H.is_wearing_item(/obj/item/clothing/suit/space/rig, slot_wear_suit)
+	R.canremove = FALSE
+	lock_atom(H)
 	R.deactivate_suit()
+	activation_animation()
+	working_animation()
 	for(var/obj/item/rig_module/RM in modules_to_install)
 		var/install_result=RM.can_install(R)
 		if(!install_result[1]) //more versatile check, allows for custom install conditions.
 			say(install_result[2], class = "binaryradio")
 			continue
+		playsound(src, 'sound/mecha/hydraulic.ogg', 40, 1)
+		spawn(rand(4 SECONDS, 5 SECONDS) / apply_multiplier)
+			playsound(src, 'sound/items/Welder.ogg', 50, 1)
 		if(do_after(H, src, 8 SECONDS / apply_multiplier, needhand = FALSE))
 			say("Installing [RM] into \the [R].", class = "binaryradio")
 			R.modules.Add(RM)
 			RM.rig = R
 			RM.forceMove(R)
 			modules_to_install.Remove(RM)
-	working_animation()
 	if(cell) //Can't answer the prompt if you're incapacitated.
 		var/choice = alert(H, "Do you wish to install [cell]?", src, "Yes", "No")
 		if((choice == "Yes") && H.Adjacent(src) && !H.incapacitated())
 			say("Installing [cell] into to \the [R].", class = "binaryradio")
-			if(R.cell)
-				R.cell.forceMove(get_turf(src))
-			cell.forceMove(R)
-			R.cell = cell
-			cell = null
-	finished_animation()
-	unlock_atom(H)
+			playsound(src, 'sound/mecha/hydraulic.ogg', 40, 1)
+			spawn(rand(4 SECONDS, 5 SECONDS) / apply_multiplier)
+				playsound(src, 'sound/misc/click.ogg', 50, 1)
+			if(do_after(H, src, 8 SECONDS / apply_multiplier, needhand = FALSE))
+				if(R.cell)
+					R.cell.forceMove(get_turf(src))
+				cell.forceMove(R)
+				R.cell = cell
+				cell = null
+	if(!R.current_glue_state)
+		R.canremove = TRUE
 	R.initialize_suit()
+	unlock_atom(H)
+	finished_animation()
 	use_power = MACHINE_POWER_USE_IDLE
 	activated = FALSE
 
 /obj/machinery/suit_modifier/proc/process_module_removal(var/mob/living/carbon/human/H)
 	if(activated)
 		return
-	lock_atom(H)
+	var/obj/item/clothing/suit/space/rig/R = H.is_wearing_item(/obj/item/clothing/suit/space/rig, slot_wear_suit)
+	if(!R)
+		return
+	// if we have something that's not a rig_module here we have a problem
+	if(!R.modules.len)
+		return
```

Comment:
Could toss a CRASH() here to force a runtime to identify this case, if it ever appears.

Reply:
This comment was not my own, it was moved here from a converted if(R.modules.len) (GIANT TABBED BLOCK). It shouldn't happen tbh and won't cause problems if it does happen (though it would be strange)